From 4de8ee2e954070011496d29eb2eb3702a427954e Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 24 Mar 2020 13:31:55 +0700
Subject: [PATCH 222/443] arm64: dts: renesas: r8a774e1: Add CPUIdle support
 for all CPU cores

This patch adds CPUIdle (Core shutdown) support for r8a774e1 SoC.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774e1.dtsi | 32 +++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774e1.dtsi b/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
index c10f0a7..dfd9ea7 100644
--- a/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
@@ -145,6 +145,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z>;
 			operating-points-v2 = <&cluster0_opp>;
+			cpu-idle-states = <&CPU_SLEEP_0>;
 		};
 
 		a57_1: cpu@1 {
@@ -156,6 +157,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z>;
 			operating-points-v2 = <&cluster0_opp>;
+			cpu-idle-states = <&CPU_SLEEP_0>;
 		};
 
 		a57_2: cpu@2 {
@@ -167,6 +169,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z>;
 			operating-points-v2 = <&cluster0_opp>;
+			cpu-idle-states = <&CPU_SLEEP_0>;
 		};
 
 		a57_3: cpu@3 {
@@ -178,6 +181,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z>;
 			operating-points-v2 = <&cluster0_opp>;
+			cpu-idle-states = <&CPU_SLEEP_0>;
 		};
 
 		a53_0: cpu@100 {
@@ -189,6 +193,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			cpu-idle-states = <&CPU_SLEEP_1>;
 		};
 
 		a53_1: cpu@101 {
@@ -200,6 +205,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			cpu-idle-states = <&CPU_SLEEP_1>;
 		};
 
 		a53_2: cpu@102 {
@@ -211,6 +217,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			cpu-idle-states = <&CPU_SLEEP_1>;
 		};
 
 		a53_3: cpu@103 {
@@ -222,6 +229,7 @@
 			enable-method = "psci";
 			clocks = <&cpg CPG_CORE R8A774E1_CLK_Z2>;
 			operating-points-v2 = <&cluster1_opp>;
+			cpu-idle-states = <&CPU_SLEEP_1>;
 		};
 
 		L2_CA57: cache-controller-0 {
@@ -237,6 +245,30 @@
 			cache-unified;
 			cache-level = <2>;
 		};
+
+		idle-states {
+			entry-method = "psci";
+
+			CPU_SLEEP_0: cpu-sleep-0 {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x0010000>;
+				local-timer-stop;
+				entry-latency-us = <400>;
+				exit-latency-us = <500>;
+				min-residency-us = <4000>;
+				status = "okay";
+			};
+
+			CPU_SLEEP_1: cpu-sleep-1 {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x0010000>;
+				local-timer-stop;
+				entry-latency-us = <700>;
+				exit-latency-us = <700>;
+				min-residency-us = <5000>;
+				status = "okay";
+			};
+		};
 	};
 
 	extal_clk: extal {
-- 
2.7.4

